### PR TITLE
refactor: Move scheme eval.

### PR DESCRIPTION
**Pull request by prmukherj**

### CODE REVIEW

File: src/ansys/fluent/core/session_solver.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def settings(self):
                 flproxy=self._settings_service,
                 version=self._version,
                 file_transfer_service=self._file_transfer_service,
-                scheme_eval=self._fluent_connection.scheme_eval.scheme_eval,
+                scheme_eval=self._fluent_connection._connection_interface.scheme_eval.scheme_eval,
```
Comment (mkundu1) [inline src/ansys/fluent/core/session_solver.py:217]: I guess at session level, we have `self.scheme_eval` directly

File: src/ansys/fluent/core/launcher/pyfluent_enums.py [hunk 2]
```diff
@@ -164,7 +164,9 @@ def _get_running_session_mode(
         try:
             session_mode = FluentMode.get_mode(
                 "solver"
-                if fluent_connection.scheme_eval.scheme_eval("(cx-solver-mode?)")
+                if fluent_connection._connection_interface.scheme_eval.scheme_eval(
+                    "(cx-solver-mode?)"
```
Comment (mkundu1) [inline src/ansys/fluent/core/launcher/pyfluent_enums.py:168]: Can we replace this with a public method of `_ConnectionInterface` class?

File: src/ansys/fluent/core/fluent_connection.py [hunk 3]
```diff
@@ -262,6 +265,10 @@ def get_cortex_connection_properties(self):
 
         return fluent_host_pid, cortex_host, cortex_pid, cortex_pwd
 
+    def server_exit(self):
```
Comment (mkundu1) [inline src/ansys/fluent/core/fluent_connection.py:268]: server_exit -> exit_server

Comment (prmukherj) [reply src/ansys/fluent/core/fluent_connection.py:268]: Done, Thank you.

File: src/ansys/fluent/core/launcher/pyfluent_enums.py (hunk 2 above)
Comment (prmukherj) [reply src/ansys/fluent/core/launcher/pyfluent_enums.py:168]: yes, done that. Thank you.

File: src/ansys/fluent/core/session_solver.py (hunk 1 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session_solver.py:217]: Scheme eval service is only created once at FluentConnection level only, right?

Comment (mkundu1) [reply src/ansys/fluent/core/session_solver.py:217]: yeah, but as we have added `<session>.scheme_eval` in session.py referering the same object, here we should have `self.scheme_eval`.

Comment (prmukherj) [reply src/ansys/fluent/core/session_solver.py:217]: Thank you @mkundu1. Got your point.

**Review by hpohekar — APPROVED** [review]

File: src/ansys/fluent/core/fluent_connection.py [hunk 4]
```diff
@@ -262,6 +265,14 @@ def get_cortex_connection_properties(self):
 
         return fluent_host_pid, cortex_host, cortex_pid, cortex_pwd
 
+    def is_solver_mode(self):
+        """Returns ``True`` if the Fluent session is in solver mode."""
+        return self.scheme_eval.scheme_eval("(cx-solver-mode?)")
+
+    def exit_server(self):
+        """Scheme call to exit the server."""
```
Comment (PipKat) [inline src/ansys/fluent/core/fluent_connection.py:273]: Start description with a verb ending in "s" or "es"

File: src/ansys/fluent/core/fluent_connection.py [hunk 5]
```diff
@@ -262,6 +265,19 @@ def get_cortex_connection_properties(self):
...
+        """Checks if the Fluent session is in solver mode.
+
+        Returns
+        --------
+            ``True`` if the Fluent session is in solver mode, ``False`` otherwise.
+        """
+        return self.scheme_eval.scheme_eval("(cx-solver-mode?)")
+
+    def exit_server(self):
+        """Scheme call to exit the server."""
```
Comment (prmukherj) [inline src/ansys/fluent/core/fluent_connection.py:278]:
```suggestion
        """Exits the server."""
```

File: src/ansys/fluent/core/fluent_connection.py (hunk 4 above)
Comment (prmukherj) [reply src/ansys/fluent/core/fluent_connection.py]: Thank you. Done.

File: src/ansys/fluent/core/fluent_connection.py [hunk 6]
```diff
@@ -225,9 +225,12 @@ def _get_channel(ip: str, port: int):
     )
 
 
-class _ConnectionProperties:
-    def __init__(self, scheme_eval):
-        self.scheme_eval = scheme_eval
+class _ConnectionInterface:
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/fluent_connection.py:228]: @prmukherj rename `fluent_build_info()` -> `product_build_info()`?

File: src/ansys/fluent/core/session.py [hunk 7]
```diff
@@ -124,7 +124,7 @@ def build_from_fluent_connection(
         self._fluent_connection = fluent_connection
         self._file_transfer_service = file_transfer_service
         self._error_state = fluent_connection._error_state
-        self.scheme_eval = fluent_connection.scheme_eval
+        self.scheme_eval = fluent_connection._connection_interface.scheme_eval
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/session.py:127]: `scheme_eval` can be an independent construction argument.

File: src/ansys/fluent/core/fluent_connection.py (hunk 6 above)
Comment (prmukherj) [reply src/ansys/fluent/core/fluent_connection.py:228]: Done, Thank you.

File: src/ansys/fluent/core/session.py (hunk 7 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session.py:127]: @seanpearsonuk, should this be done now or during next level of refactoring when we touch session. The plan was to keep this PR specific to FluentConnection.

Comment (seanpearsonuk) [reply src/ansys/fluent/core/session.py:127]: I will leave it up to you.

Comment (prmukherj) [reply src/ansys/fluent/core/session.py:127]: Sure, I'll look into it.

**Pull request merged**